### ISSUE TITLE
Webpack doesn't resolve not ignored node_modules

### ISSUE THREAD

**Issue by isnifer** [webpack/webpack#7856]
```json
{
  "webpack": "4.5.0",
  "webpack-cli": "2.0.14",
  "webpack-dev-server": "3.1.4",
  "webpack-hot-client": "2.2.2"
}
```

Hi! Two years ago I started writing [babel-plugin](https://github.com/tipsi/babel-plugin-tipsi-flavors). Initially, it was tested on Metro bundler for React-Native, and it works perfectly.
Today I need to apply this plugin for my web project, and it does not work partially.

I have a project with different modules linked by Lerna:
```
projects/
  shared
  web
```
**Web depends on shared**.

The `web` has a `store.js` file. It imports network logger like this:
```js
import useLogger from 'shared/utils/useLogger'
```

Where `shared/utils/useLogger` has the next structure:
```js
// index.js
export default from './use' // <= my plugin must modify this import/export too. But it works only in Metro

// use.js
export default false

// use.logger.js
export default true
```

My plugin read every file that webpack (or metro) is transpiling.  
**But Metro resolve every imported file unlike webpack**.
Here the difference:
**Metro**:
1. import useLogger from 'shared/utils/useLogger'
2. Oh, I see there is an import — let's open this file and transpile via babel
3. *Opens `shared/utils/useLogger`*
4. Oh, I see there is a code that I can transpile — hey, babel LET'S DO IT!

**Webpack**:
1. import useLogger from 'shared/utils/useLogger'
2. Oh, I see there is an import — OK, WON'T DO ANYTHING.

That's it. Webpack does not resolve linked node_module until it'll find index.js that I need.
Is it correct behavior or not?

**Comment by montogeek** [comment]
Thanks for your report. It would be great if you reduce your issue to a small reproducible example. Best put this example into a github repository together with instructions how to get to the problem.

**Comment by isnifer** [comment]
@montogeek ok. Will do today, thank you for quick response!

**Comment by isnifer** [comment]
@montogeek I found my own error. I need to add my babel-plugin to .babelrc inside `shared` folder to activate it. Plugin added into .babelrc inside `web` is not enough.

So, I have created a repo.
1. Commit with error: https://github.com/isnifer/webpack-issue-7856/commit/603c56a28f12e667b5890eea128582346992483e
2. Commit with fix: https://github.com/isnifer/webpack-issue-7856/commit/1700f5f9587e1b19c55fb1e8e84c68b1c7167a69

Then I checked it on my own project and it works!
